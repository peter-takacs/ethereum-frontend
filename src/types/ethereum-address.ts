export class Address {
    private readonly address: string;

    public static isValid(address: string) {
        return /^0x[a-fA-F0-9]{40}$/.test(address);
    }

    public static fromBytes(bytes: Uint8Array): Address {
        if (bytes.length !== 20) {
            throw new Error(`Invalid amount of bytes supplied to ethereum address.`)
        }
        let address = '0x';
        for (const byte of bytes) {
            address = address + byte.toString(16);
        }
        return new Address(address);
    }

    constructor (address: string) {
        if (!Address.isValid(address)) {
            throw new Error(`Invalid address "${address}" supplied to ethereum address constructor.`)
        }
        this.address = address;
    }

    public toString() {
        return this.address;
    }

    public equals(other: Address | null) {
        if (other == null) {
            return false;
        }
        return this.address === other.address;
    }
}