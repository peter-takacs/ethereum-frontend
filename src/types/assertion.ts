import { Address } from "./ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}