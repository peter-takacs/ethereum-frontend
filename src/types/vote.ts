import { Address } from "./ethereum-address";

export enum VoteStatus {
    Pending = 0,
    Accept = 1,
    Reject = 2
}

export interface CandidateVotes {
    candidate: Address;
    votes: Map<Address, VoteStatus>;
}