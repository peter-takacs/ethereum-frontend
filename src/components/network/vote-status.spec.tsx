import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CandidateStatus from '../../containers/network/candidate-status';
jest.doMock('../../containers/network/candidate-status');
import { dummyAddress } from '../../utils/test/address';
import VoteStatus from './vote-status';
import { VoteStatus as VoteStatusEnum } from '../../types/vote' ;
import { Address } from '../../types/ethereum-address';

test('Renders correctly with no data', () => {
    const component = renderer.create(
            <VoteStatus
                candidateStatuses={[]}
                address={null}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
});

test('Renders correctly with data', () => {
    const votes = new Map<Address, VoteStatusEnum>();
    votes.set(dummyAddress(1), VoteStatusEnum.Accept);
    const component = renderer.create(
        <VoteStatus
            address={null}
            candidateStatuses={[
                {
                    candidate: dummyAddress(0),
                    votes
                }
            ]}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})


test('Renders correctly with an account', () => {
    const votes = new Map<Address, VoteStatusEnum>();
    votes.set(dummyAddress(1), VoteStatusEnum.Accept);
    votes.set(dummyAddress(2), VoteStatusEnum.Pending)
    const component = renderer.create(
        <VoteStatus
            address={dummyAddress(2)}
            candidateStatuses={[
                {
                    candidate: dummyAddress(0),
                    votes
                }
            ]}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})