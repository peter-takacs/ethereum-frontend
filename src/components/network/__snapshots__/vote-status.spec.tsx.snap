// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with data 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <div
      className="MuiGrid-container-1"
    >
      <p
        className="MuiTypography-root-98 MuiTypography-body1-107"
      >
        0xdf3f619804a92fdb4057192dc43dd748ea778adc
      </p>
      <div
        className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
      >
        <div
          className="MuiGrid-item-2"
        >
          0x27ecd0a598e76f8a2fd264d427df0a119903e8ea
        </div>
        <div
          className="MuiGrid-item-2"
        >
          1
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly with no data 1`] = `
<div
  className=""
/>
`;
